SharedLibrary/Sources/Shared/Grid.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "Shared/Types.hpp"
+#include "Shared/Array.hpp"
+#include <vector>
+#include <functional>
+
+namespace uf {
+
+struct GridTransformation {
+	vec3i shift = {0,0,0};
+	vec3i resize = {0,0,0};
+};
+
+template<typename T>
+class Grid {
+public:
+	Grid() {};
+	void SetSize(vec3u size);
+	void SetSize(uint x, uint y, uint z);
+	vec3u GetSize() const;
+	std::vector<T>& Get();
+	const std::vector<T>& Get() const;
+	void Transform(GridTransformation transformation, std::function<void(vec3i, vec3i)> update, std::function<void(vec3i)> remove);
```

Comment:
There isn't obvious when callbacks will be called and what arguments they receive.

I propose to name them movedElementsProcessor and removedElementsProcessor.